### DISCUSSION TITLE
Welcome to scancode.io Discussions!

### DISCUSSION THREAD

**Discussion by tdruez** [nexB/scancode.io#255] [Q&A] [answered]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you'll use this space to:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others with an open-minded attitude. Remember that this is a community we build together 💪.

  To get started, comment below with an introduction of yourself and tell us why you joined this community.

**Comment by Khan-Tzar** [comment]
Hello, 
Could you consider CPU, Memory container limitations into docker-compose.yml , please?
Something like this: 

```bash
web:
    build: .
    command: sh -c "
        ./manage.py migrate &&
        ./manage.py collectstatic --no-input --verbosity 0 --clear &&
        gunicorn scancodeio.wsgi:application --bind :8000 --timeout 600 --workers 8"
    env_file:
      - docker.env
    expose:
      - 8000
    volumes:
      - /etc/scancodeio/:/etc/scancodeio/
      - workspace:/var/scancodeio/workspace/
      - static:/var/scancodeio/static/
    depends_on:
      - db
    restart: always
    deploy:
        resources:
            limits:
                cpus: 2
                memory: 4096m
            reservations:
                memory: 512m
```

**Reply by tdruez** [reply]
Hi @Khan-Tzar, could you provide more context about your issues and the benefits of forcing those limitations?

**Comment by 19csa55** [comment]
Hello,

I'm a newbie, and as I've understood, while starting a new project, the Docker URL provided points to Docker Hub and all the images uploaded there. 

I was wondering if I could provide a URL for referring to an image in another private image repository, hosted in a private network, such as Harbor. 

Harbor is private image repository that can be hosted and managed. Is it possible to use Scancode.io with that?

P.S. I think I've disturbed an ancient thread. Sorry in advance!

**Reply by tdruez** [reply]
Hello @19csa55, private repos are not currently supported but we'd welcome a PR if you want to give it a try.

ScanCode.io is using skopeo to fetch docker images. Would this type of command: https://github.com/containers/skopeo?tab=readme-ov-file#using---creds-directly work with your context?

**Reply by 19csa55** [reply]
Hi tdruez, 

Yes, that command will work, took a while to get it all running in my environment!

So how do we get started adding this to a working Scancode.io instance?

**Reply by tdruez** [reply]
You can look at some ideas we discussed in the past at https://github.com/nexB/scancode.io/issues/203#issuecomment-933299906

**Reply by 19csa55** [reply]
Hi tdruez,

Was looking at the code, trying to understand it step by step.

Can we pm if possible? I do not understand the #203 idea and how to setup a new SKOPEO_CREDENTIALS setting.

**Reply by pombredanne** [reply]
@19csa55 we prefer using public discussions rather than private messages unless this is a private matter  and I doubt this is the cases here.
In short:
- we run skopeo to fetch container images
- to access private container registries we would need to provide authentication. One design is to store auth in some settings and/or envt. var. There may be other ways though that should be explored, like the  standard docker auth JSON file.

Can you start by installing and playing with skopeo?

**Reply by 19csa55** [reply]
Hello @pombredanne 

Ah, well thanks for the quick response anyways!

Can you tell me more about storing auth settings? I have even tried to change the skopeo command being used in /scanpipes/pipes/fetch.py to add my creds and make it point to the Harbor registry that I'm using, but it seems that it causes an error in the run_command_safely() function where a subprocess is used.

**Reply by 19csa55** [reply]
@pombredanne Hello.

I've gotten a bit of it working now, is it possible for you to guide me through any troubleshooting procedures while the docker image is running? It's difficult for me to look at the output of the subprocess commands.

**Reply by tdruez** [reply, marked as answer]
Support for private Docker registry added in https://github.com/nexB/scancode.io/pull/1098
See https://scancodeio.readthedocs.io/en/latest/faq.html#how-to-fetch-files-from-private-sources-and-protected-by-credentials and https://scancodeio.readthedocs.io/en/latest/application-settings.html#scancodeio-skopeo-credentials

@19csa55 could you give it a try and let me know it that approach works for you?

**Reply by 19csa55** [reply]
Was told to look back into this and it now works, thanks for your help!

**Comment by grewalrehmat** [comment]
hey guys 
I am a newbie in Open Source having proficiency in django, python, cpp, relational databases  and i'm trying to become a GSoC contributor will someone be kind enough to guide me 
thanks a lot

**Comment by rajputdivyanshu81** [comment]
Hello Everyone 
This is Divyanshu Rajput, starting my journey to contribute to Aboutcode
Needs guidance regarding project setup, not able to find documentation to locally set up the project

**Comment by Monal-Reddy** [comment]
Hello, I am Monal, a third year Btech student. I am interested in contributing to ScanCode.io. I have started setting up the environment and am currently going through the documentation to understand how the pipelines and components work. I would love guidance on good beginner issues and areas where contributions are currently needed. I look forward to contributing and learning more

**Comment by hemant-rgb** [comment]
Hi maintainers 👋,

I'm Hemant, a second-year student and a new contributor interested in consistently contributing to ScanCode.io.

I've successfully set up the full development environment (Docker + backend + pipelines) and started exploring the codebase.
This PR is my first contribution, and I’m working on more improvements already.

I would love to continue contributing to ScanPipe and the broader AboutCode ecosystem.
Please let me know if there are any recommended beginner-friendly areas, tasks, or files that would be most helpful for me to focus on next.

Thanks for creating such a well-structured project — excited to contribute more! 🙌

— Hemant Machiwar(hemantmachiwar76@gmail.com)

**Comment by SiminJi1005** [comment]
Hi maintainers, I'm Simin, I am currently an MS in Computer Science student at NEU, and I am very interested in applying to GSoC 2026 with AboutCode.

My background is mainly in Python, Java, backend development, and web applications. I am especially interested in ScanCode.io because it combines backend engineering, automation, and practical open source compliance/security workflows.

At the moment, I am exploring project directions around ScanCode.io and would appreciate guidance on:

which ideas are a good fit for a new contributor with strong backend/web interest,
which repository/issues would be the best place to start contributing,
whether there are any recommended beginner-friendly issues or proposal expectations I should review first.

I am currently reading the documentation and prior GSoC project reports, and I plan to start with a small contribution while preparing my proposal draft.

Thank you, and I’d be grateful for any guidance.

**Reply by SiminJi1005** [reply]
After reviewing the 2026 project ideas, I am currently leaning toward the ScanCode.io project on migrating vulnerability fetching to the new advisories-based API from VulnerableCode. It looks like a strong fit for my backend/API interests, and I plan to study the related issue and code paths next.
